ena-submission/scripts/trigger_submission_to_ena.py

```diff
@@ -103,19 +103,25 @@ def trigger_submission_to_ena(
             return
 
     while True:
+        logger.debug("Checking for new sequences to upload to submission_table")
         # In a loop get approved sequences uploaded to Github and upload to submission_table
-        response = requests.get(
-            config.github_url,
-            timeout=10,
-        )
-
-        if response.ok:
+        try:
+            response = requests.get(
+                config.github_url,
+                timeout=10,
+            )
+            response.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Failed to retrieve file due to requests exception: {e}")
+            time.sleep(min_between_github_requests * 60)
+            continue
+        try:
             sequences_to_upload = response.json()
-        else:
-            error_msg = f"Failed to retrieve file: {response.status_code}"
-            logger.error(error_msg)
-        upload_sequences(db_config, sequences_to_upload)
-        time.sleep(min_between_github_requests * 60)  # Sleep for x min to not overwhelm github
+            upload_sequences(db_config, sequences_to_upload)
+        except Exception as upload_error:
+            logger.error(f"Failed to upload sequences: {upload_error}")
+        finally:
```

Comment:
I think the finally is unnecessary because if an exception is thrown we won't need to sleep anyways

Reply:
I think its good to keep it in, if we for what ever weird reason can't upload to the db we should probably rate limit a bit not to overwhelm github or the database even more.

Reply:
In my understanding, if we throw an uncaught exception, it will exit the loop anyways - but yeah maybe it restarts then? Doesn't really matter, so happy to let go, just noticed when I read the code and was wondering.